File: src/main/java/space/ryzhenkov/Fabric2Discord/mixins/PlayerManagerMixin.java
package space.ryzhenkov.Fabric2Discord.mixins;

import net.minecraft.network.ClientConnection;
import net.minecraft.server.PlayerManager;
import net.minecraft.server.network.ServerPlayerEntity;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
import space.ryzhenkov.Fabric2Discord.config.F2DConfig;
import space.ryzhenkov.Fabric2Discord.utils.MessageUtils;

@Mixin(PlayerManager.class)
public abstract class PlayerManagerMixin {
    @Inject(method = "onPlayerConnect", at = @At("HEAD"))
    private void onPlayerConnect(ClientConnection connection, ServerPlayerEntity player, CallbackInfo ci) {
        if (!F2DConfig.messages.playerJoin.INSTANCE.getEnabled()) return;
        if (F2DConfig.general.ids.INSTANCE.getLogChannel() == null) return;
        MessageUtils.INSTANCE.sendEmbedMessage(F2DConfig.general.ids.INSTANCE.getLogChannel(),
                MessageUtils.INSTANCE.getConfigMessage(F2DConfig.messages.playerJoin.INSTANCE, null, player, null)
                        .build().asRequest()
        );
    }
}

File: src/main/kotlin/space/ryzhenkov/fabric2discord/ktmixins/PlayerManagerMixin.kt
package space.ryzhenkov.fabric2discord.ktmixins

import net.minecraft.server.network.ServerPlayerEntity
import space.ryzhenkov.fabric2discord.config.ConfigAPI
import space.ryzhenkov.fabric2discord.utils.MessageUtils

object PlayerManagerMixin {
    fun onPlayerConnect(player: ServerPlayerEntity) {
        if (!ConfigAPI.messages.playerJoin.enabled) return
        if (ConfigAPI.general.ids.getLogChannelOrNull() == null) return
        MessageUtils.sendEmbedMessage(ConfigAPI.general.ids.getLogChannelOrNull()) {
            MessageUtils.getEmbedMessage(ConfigAPI.messages.playerJoin, player, null)
        }
    }
}